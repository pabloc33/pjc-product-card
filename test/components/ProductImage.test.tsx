import React from 'react';
import render from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('debe mostrar el componente correctamente con la imagen personalizada ', () => {
    const wrapper = render.create(
      <ProductImage img="https://png.pngtree.com/element_our/20200702/ourmid/pngtree-vector-pattern-material-click-url-image_2281151.jpg" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe mostrar el componente con la imagen del producto ', () => {
    const wrapper = render.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
