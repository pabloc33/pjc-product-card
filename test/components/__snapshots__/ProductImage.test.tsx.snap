// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente con la imagen del producto  1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product imag"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe mostrar el componente correctamente con la imagen personalizada  1`] = `
<img
  alt="Product imag"
  className="productImg undefined"
  src="https://png.pngtree.com/element_our/20200702/ourmid/pngtree-vector-pattern-material-click-url-image_2281151.jpg"
/>
`;
